### PR TITLE
Add testing documentation

### PR DESCRIPTION
**Pull request by QuantumEnigmaa**
Testing guidelines

### CODE REVIEW

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+* Deploy the new Promtail version using opsctl (replace `test-branch` with your branch's name): `opsctl deploy --use-kubeconfig promtail-app@test-branch`
+
+### The "hard" way
+
+If for some reason you can't use the `opsctl deploy` command or don't want to:
+* Either create a draft PR or directly use your branch
+* Look at the build step `ci/circleci: package and push promtail chart`, step `Push chart archive to OCI registry app catalog`
+    * Take note of pushed image name
+* Edit the `promtail` app: `kubectl edit -n giantswarm promtail`
```
Comment (marieroque) [inline TESTING.md:18]: `kubectl edit -n giantswarm promtail` => `kubectl edit app -n giantswarm promtail`

File: TESTING.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* Update catalog name and image version according to the image name shown in build logs
+* Make sure promtail is deployed with the right version: `kubectl get app -n giantswarm promtail`
+
+## Make sure Promtail is scraped by prometheus
+
+* Open prometheus on the MC from the installation: `opsctl open -i gauss -a pometheus --workload-cluster gauss`
```
Comment (marieroque) [inline TESTING.md:24]: `pometheus` => `prometheus`

**Review by marieroque — COMMENTED** [review]
Just typo remarks

File: TESTING.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+* Open the grafana instance from gorilla: `opsctl open -i gorilla -a grafana`
+* Go to the *Explore* section and select `Loki` as datasource.
+* Execute a query based on the following example: `sum(count_over_time({installation="gauss", cluster_id="gauss", hostname=~"ip-.*.eu-west-1.compute.internal"}[10m])) by (hostname)`
+* You should see as many different colors as there are nodes on the targeted cluster (8 for gauss). Take a screenshot for the PR.
+* Execute another query based on this example: `{installation="gauss", cluster_id="gauss", pod=~"k8s-api-server-ip-.*"}`
+* You should see logs from the api-server pods on the targeted cluster. Take a screenshot for the PR.
+
+## Clean environment
+
+* Once you're finished testing, simply resume flux: `flux resume kustomization -n flux-giantswarm collection; flux resume kustomization -n flux-giantswarm flux`. (You might consider not resuming flux when working on gauss as many users might need it to stay suspended)
```
Comment (QuentinBisson) [inline TESTING.md:41]:
```suggestion
* Once you're finished testing, simply resume flux: `flux resume kustomization -n flux-giantswarm collection; flux resume kustomization -n flux-giantswarm flux`. (You might consider not resuming flux when working on test installations as many users might need it to stay suspended)
```

File: TESTING.md (hunk 2 above)
Comment (QuantumEnigmaa) [reply TESTING.md]: Done

File: TESTING.md (hunk 1 above)
Comment (QuantumEnigmaa) [reply TESTING.md]: I found myself many times in a situation where I had no choice but using the "hard way" to install the new version. That's why I think it's a good think to have it written for anyone who might encounter the same issue

Comment (QuentinBisson) [reply TESTING.md]: I'm fine with the curretn structure but I'd like if we have the same base template ;)

File: TESTING.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+* Suspend these 2 flux kustomizations:`flux suspend kustomization -n flux-giantswarm collection; flux suspend kustomization -n flux-giantswarm flux`
+
+### The "easy" way
+
+* Deploy the new Promtail version using opsctl (replace `test-branch` with your branch's name): `opsctl deploy --use-kubeconfig promtail-app@test-branch`
+
+### The "hard" way
+
+If for some reason you can't use the `opsctl deploy` command or don't want to:
+* Either create a draft PR or directly use your branch
```
Comment (hervenicol) [inline TESTING.md:15]: Does the "easy" way work without that? Easy way also relies on the apps pushed to the test catalogs, and AFAIK this is done on PR. So I'd say this line goes before chosing between easy/hard.

File: TESTING.md (hunk 1 above)
Comment (QuentinBisson) [reply TESTING.md]: That would be nice yes :)

File: TESTING.md (hunk 4 above)
Comment (QuantumEnigmaa) [reply TESTING.md:15]: True, moving this

File: TESTING.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
+# How to test
+
+## Install the new Promtail version
+
+* Choose a test cluster: `opsctl login gauss`
+* Suspend these 2 flux kustomizations:`flux suspend kustomization -n flux-giantswarm collection; flux suspend kustomization -n flux-giantswarm flux`
```
[the remark this answers is not in the document]
Comment (QuantumEnigmaa) [reply TESTING.md:6]: Ok, I had a doubt about that

File: TESTING.md (hunk 1 above)
Comment (QuantumEnigmaa) [reply TESTING.md]: Created a PR here : https://github.com/giantswarm/giantswarm/pull/25200

**Comment by QuantumEnigmaa** [comment]
Instead of writing the entire "hard way" method I put a link to the intranet page created with the following PR: https://github.com/giantswarm/giantswarm/pull/25200

**Review by QuentinBisson — APPROVED** [review]

**Comment by QuentinBisson** [comment]
Great doc :)

**Pull request merged**